lab_1_keywords_tfidf/main.py

```diff
@@ -3,103 +3,157 @@
+            if not isinstance(i, check_value):
+                return False
+    return True
+
+
+def check_num(num_arg):
+    if not isinstance(num_arg, int) or type(num_arg) == bool:
+        return False
+    if num_arg < 0:
+        return False
+    return True
 
 
 def clean_and_tokenize(text: str) -> Optional[list[str]]:
     """
     Removes punctuation, casts to lowercase, splits into tokens
-
     Parameters:
     text (str): Original text
-
     Returns:
     list[str]: A sequence of lowercase tokens with no punctuation
-
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not check_input_type(text, str):
+        return None
+    text_small = text.lower().strip()
+    tokens_str = ''
+    for i in text_small:
+        if i.isalnum() is True or i == ' ' or i == '\n':
+            tokens_str += i
+    tokens = tokens_str.split()
+    return tokens
 
 
 def remove_stop_words(tokens: list[str], stop_words: list[str]) -> Optional[list[str]]:
     """
     Excludes stop words from the token sequence
-
     Parameters:
     tokens (List[str]): Original token sequence
     stop_words (List[str]: Tokens to exclude
-
     Returns:
     List[str]: Token sequence that does not include stop words
-
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not check_input_type(tokens, list, str, False) or \
+            not check_input_type(stop_words, list, str):
+        return None
+    no_stop_words = []
+    for x in tokens:
+        if x not in stop_words:
+            no_stop_words.append(x)
+        else:
+            continue
+    return no_stop_words
 
 
 def calculate_frequencies(tokens: list[str]) -> Optional[dict[str, int]]:
     """
     Composes a frequency dictionary from the token sequence
-
     Parameters:
     tokens (List[str]): Token sequence to count frequencies for
-
     Returns:
     Dict: {token: number of occurrences in the token sequence} dictionary
-
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not check_input_type(tokens, list, str, False):
+        return None
+    tokens_freqs = {}
+    for i in tokens:
+        tokens_freqs[i] = tokens.count(i)
+    return tokens_freqs
 
 
 def get_top_n(frequencies: dict[str, Union[int, float]], top: int) -> Optional[list[str]]:
     """
     Extracts a certain number of most frequent tokens
-
     Parameters:
     frequencies (Dict): A dictionary with tokens and
     its corresponding frequency values
     top (int): Number of token to extract
-
     Returns:
     List[str]: Sequence of specified length
     consisting of tokens with the largest frequency
-
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not check_input_type(frequencies, dict, str, Union[int, float], False) \
+            or not check_num(top):
+        return None
+    sorted_tokens_freqs = {k: v for k, v in sorted(frequencies.items(), key=lambda item: item[1], reverse=True)}
```

Comment:
dictionaries as a type do not really have any specific order within it, in contrast to lists
so it would be better to compose a list from the sorted items right away instead of making a new dict and only then converting its keys to a list